### PR TITLE
API Implementation

### PR DESCRIPTION
**Pull request by ogiogiovictor**
## Project Requirements

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Add authentication to your API endpoints.
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Api::V1::UsersController < ApplicationController
+  before_action :find_user, except: %i[index]
```
Comment (danielufeli) [inline app/controllers/api/v1/users_controller.rb]: - [ ] Nice working on your project, kindly check I get the error in the screenshot below when trying to login using postman.
image

**Review by danielufeli — CHANGES_REQUESTED** [review]
Hi @Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlmostThereReachGIF

## Highlights
- Nice commits messages✔️
- Descriptive PR ✔️
- Clean Code ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/concerns/response.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module Response
+  def json_response(object, status = :ok)
+    render json: object, status:
+  end
+end
```
Comment (Munsa1) [inline app/controllers/concerns/response.rb:5]: - [x] Kindly fix the syntax error in this file.
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @ogiogiovictor, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/concerns/response.rb (hunk 2 above)
Comment (ogiogiovictor) [reply app/controllers/concerns/response.rb:5]: Dear Munsa1,

I don't have that error in my code base. You might want to check your end. What we are testing for is the API and this is my response from postman.

image

image

Comment (ogiogiovictor) [reply app/controllers/concerns/response.rb:5]: please pull the repo again.

Comment (ogiogiovictor) [reply app/controllers/concerns/response.rb:5]: Fixed error everything works fine @Munsa1 @danielufeli

**Review by whiz25 — CHANGES_REQUESTED** [review]
Hi @ogiogiovictor, 

Good job so far! 
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🎉 🎉 🎉 
- Changes made as requested

## Required Changes ♻️
_Check the comments under the review._
- Please fix the litter errors

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi team 👋, 

Well done for making all the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**